### ISSUE TITLE
[Feature request]: package to use jwt for backend

### ISSUE THREAD

**Issue by landsman** [supabase-community/supabase-kt#415]
### General Info

- [X] I installed the latest version of Supabase Kt
- [X] I checked for similar feature requests

### Feature request

Please provide a package to decode JWT auth. Including Supabase user interface etc.
It would be lovely to have ready-made integration for the most common Spring Boot Framework (Java/Kotlin).

Inspiration: https://github.com/tschuehly/htmx-supabase-spring-boot-starter/tree/master/src/main/kotlin/de/tschuehly/htmx/spring/supabase/auth

### Usecase

Typically you have React / Next.js application for frontend and backend in Spring Boot. So you want to get a Supabase access token and verify it on the backend to let the user in and pair it to backend entities.

- https://github.com/tschuehly/htmx-supabase-spring-boot-starter
- https://github.com/tschuehly/demo-supabase-spring-boot-starter

**Comment by jan-tennert** [comment]
Could you clarify what exactly such a library should provide? So JWT decoding and verification?

**Comment by nateweisz** [comment]
> Could you clarify what exactly such a library should provide? So JWT decoding and verification?

I'd also be interested in this for ktor as a server. Our frontend uses supabase for authentication and I need to be able to intercept requests and check for authentication stuff.

**Comment by jan-tennert** [comment]
> > Could you clarify what exactly such a library should provide? So JWT decoding and verification?
> 
> I'd also be interested in this for ktor as a server. Our frontend uses supabase for authentication and I need to be able to intercept requests and perform authentication checks on the backend.

So it should work like a ktor plugin which automatically checks/rejects requests based on the JWT?

**Comment by nateweisz** [comment]
> > > Could you clarify what exactly such a library should provide? So JWT decoding and verification?
> > 
> > 
> > I'd also be interested in this for ktor as a server. Our frontend uses supabase for authentication and I need to be able to intercept requests and perform authentication checks on the backend.
> 
> So it should work like a ktor plugin which automatically checks/rejects requests based on the JWT?

Yes, or at least makes it possibly to build that into your system without too much code.

**Comment by jan-tennert** [comment]
Alright, if anyone wants to try to take this on, feel free to ask me for questions for integrating this into supabase-kt. I think ideally there is some basic JWT stuff inside gotrue-kt and then there are spring-boot/ktor modules inside /plugins (like Compose Auth Ui which is not really a supabase-kt plugin)

**Comment by landsman** [comment]
@tschuehly can you help us there? I was able to use your code from mentioned repository and I think that "official" integration would be amazing for Spring Framework ecosystem :)

**Comment by tschuehly** [comment]
@landsman 
What kind of help would you need? You should be able to use the htmx-supabase-starter also in non htmx spring boot projects.

It would probably make sense to partition the different functionality.

**Comment by landsman** [comment]
@tschuehly I would appreciate your feedback on what would be good to provide (classes, methods) from `supabase-kt` to be easy to use in projects and packages like yours.

**Comment by nateweisz** [comment]
> Alright, if anyone wants to try to take this on, feel free to ask me for questions about integrating this into supabase-kt. I think ideally there is some basic JWT stuff inside gotrue-kt and then there are spring-boot/ktor modules inside /plugins (like Compose Auth Ui which is not really a supabase-kt plugin). I'm neither an expert in spring-boot nor server side authentication with ktor so it might take a while until I start to work on it.

I'm currently working on a backend that will use it. I can try and post some code/work on it once I get that done but mostly waiting on my frontend team to get started.
